### DISCUSSION TITLE
Option to allow file transfer from stable to lazer to use all system resources...

### DISCUSSION THREAD

**Discussion by Raregendary** [ppy/osu#27298] [Ideas] [answered]
**Please add option to use all system resources to speed up the transfer from stable to lazer significantly**
I have 7800x3d processor and my SSD does 6 GB/s sequential _(both games are on it and the OS)_, and yet i look at osu lazer using 1 core and transferring with like 50 MB/s my maps waiting for years for it to finish.  I guess my problem for having over 150 GB of songs but i bet im not the only one.

**Comment by Raregendary** [comment]
I wrote the post when i started the transfer, i think there is memory leak during song import but after over 2 hours it reached 3GB and crashed being at 11/12k and now i gotta wait additionally to do the slowest check if i have the map ever with like barely 30 ish a second, that aswell needs fixing...

**Reply by peppy** [reply]
3gb is expected memory usage, especially if you are importing weird beatmaps like "aspire" stuff. If you got a crash then please attach logs.

**Reply by Raregendary** [reply]
Sure here are the logs
Logs saved when it crashed:
[logs.zip](https://github.com/ppy/osu/files/14368399/logs.zip)

Windows event viewer logs:
[Application Error.txt](https://github.com/ppy/osu/files/14368443/Application.Error.txt)
[dotNET Runtime.txt](https://github.com/ppy/osu/files/14368444/dotNET.Runtime.txt)

All logs:
[logs.zip](https://github.com/ppy/osu/files/14368404/logs.zip)

Wait there is an export logs... i forgor
[compressed-logs.zip](https://github.com/ppy/osu/files/14368452/compressed-logs.zip)

**Reply by peppy** [reply]
Crash is native in realm:

```
Application: osu!.exe CoreCLR Version: 8.0.224.6711 .NET Version: 8.0.2 Description: The process was terminated due to an unhandled exception. Exception Info: System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt. Stack: at Realms.ListHandle+NativeMethods.find_value(Realms.ListHandle, Realms.Native.PrimitiveValue, Realms.NativeException ByRef) at Realms.ListHandle+NativeMethods.find_value(Realms.ListHandle, Realms.Native.PrimitiveValue, Realms.NativeException ByRef) at Realms.RealmList`1[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].IndexOf(System.__Canon) at Realms.RealmCollectionBase`1[[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].Contains(System.__Canon) at osu.Game.Screens.Select.Carousel.CarouselBeatmap.checkMatch(osu.Game.Screens.Select.FilterCriteria) at 
```

**Reply by Raregendary** [reply]
I have no idea what that means but i had the exact same error 23 hours earlier before that one. 
_But i bet im not the one who protected that memory causing it to crash_  😉

**Reply by peppy** [reply]
It's a crash in a third party library we use, a low level c++ issue. It will need to be reported to them, although it's going to be very hard to track down unless a reliable repro is created.

This is the first we've heard of it, so it may be specific to your hardware / OS / software, as well.

I've reported over at https://github.com/realm/realm-core/issues/7366.

**Reply by peppy** [reply]
@Raregendary could you please provide your `client.realm` file zipped? I need to try and create a reproduction case for realm to fix the bug.

**Comment by peppy** [comment, marked as answer]
It's slow because of online metadata fetching. This will eventually be improved.
